rest/health.go

```diff
@@ -102,124 +67,49 @@ func (h *healthResource) webService(basePath string) *restful.WebService {
 }
 
 func (h *healthResource) check(request *restful.Request, response *restful.Response) {
-	type chanResult struct {
-		name string
-		HealthResult
-	}
-
 	var (
 		service = request.QueryParameter("service")
-		result  = HealthResponse{
-			Status:   HealthStatusHealthy,
-			Message:  "",
-			Services: map[string]HealthResult{},
-		}
-
-		resultChan = make(chan chanResult)
-		once       sync.Once
+		ctx     = request.Request.Context()
+		code    = http.StatusOK
 	)
-	defer once.Do(func() { close(resultChan) })
-
-	ctx := request.Request.Context()
-	g, _ := errgroup.WithContext(ctx)
-
-	for name, healthCheck := range h.healthChecks {
-		name := name
-		healthCheck := healthCheck
-
-		g.Go(func() error {
-			if h == nil {
-				return nil
-			}
-			if service != "" && name != service {
-				return nil
-			}
-
-			result := chanResult{
-				name: name,
-				HealthResult: HealthResult{
-					Status:   HealthStatusHealthy,
-					Message:  "",
-					Services: map[string]HealthResult{},
-				},
-			}
-			defer func() {
-				resultChan <- result
-			}()
-
-			var err error
-			result.HealthResult, err = healthCheck.Check(ctx)
-			if err != nil {
-				result.Message = err.Error()
-				h.log.Error("unhealthy service", "name", name, "status", result.Status, "error", err)
-			}
-
-			return err
-		})
-	}
-
-	finished := make(chan bool)
-	go func() {
-		for r := range resultChan {
-			r := r
-			result.Services[r.name] = r.HealthResult
-		}
-		finished <- true
-	}()
 
-	rc := http.StatusOK
+	result, err := h.healthCheck.Check(ctx)
+	if err != nil {
+		h.log.Error("unhealthy application", "status", result.Status, "error", err)
 
-	if err := g.Wait(); err != nil {
-		rc = http.StatusInternalServerError
-		result.Message = err.Error()
+		code = http.StatusInternalServerError
+		if result.Status == healthstatus.HealthStatusHealthy {
```

Comment:
Was this in the code before? To me it looks a bit suspicious because we derive degraded status from an error. Maybe it's more natural to just maintain the actual result and do the "DeriveOverallStatus" thing in case the `Status` field is empty.

Reply:
This wasn't in the old code and I added it. Your proposed change is definitely the way to go